plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java

```diff
@@ -695,6 +704,34 @@ public void renameSchema(ConnectorSession session, String schemaName, String new
         return properties.buildOrThrow();
     }
 
+    private static LongReadFunction dateReadFunction()
+    {
+        return (resultSet, index) -> {
+            // Convert to LocalDate from java.sql.Date via String because java.sql.Date#toLocalDate() returns wrong results in B.C. dates. -5881579-07-11 -> +5881580-07-11
```

Comment:
Is this a problem in all JDBC based connectors, or specifically in Phoenix?

Reply:
It's not specific to Phoenix.

Reply:
So the followup question... Is that something to fix in base jdbc, or fix it for each driver?

Reply:
The method is already deprecated. I wouldn't modify in this situation and would fix each connectors.